src/Peddler/DoubleGenerator.cs

```diff
@@ -0,0 +1,237 @@
+using System;
+using System.Collections.Generic;
+using System.Threading;
+
+namespace Peddler {
+    /// <summary>
+    ///   An implementation of <see cref="IDistinctGenerator{Double}" />.
+    /// </summary>
+    /// <inheritdoc />
+    public class DoubleGenerator : IComparableGenerator<Double> {
+        private static ThreadLocal<Random> random { get; } =
+            new ThreadLocal<Random>(() => new Random());
+
+        private const Int32 EpsilonExponent = -323;
+        private const Int32 MinimumIntervalExponentDelta = 16;
+
+        /// <summary>
+        /// The minimum value to generate in calls to <see cref="Next()" />. Defaults to
+        /// <see cref="Double.MinValue" />
+        /// </summary>
+        public Double Minimum { get; }
+
+        /// <summary>
+        /// The maximum value to generate in calls to <see cref="Next()" />. Defaults to
+        /// <see cref="Double.MaxValue" />.
+        /// </summary>
+        public Double Maximum { get; }
+
+        private Int32 defaultMinimumExponent { get; }
+        private Int32 defaultMaximumExponent { get; }
+        private List<(Int32, Int32)> defaultScales { get; }
+
+        /// <inheritdoc />
+        public IEqualityComparer<Double> EqualityComparer => EqualityComparer<Double>.Default;
+
+        /// <inheritdoc />
+        public IComparer<Double> Comparer => Comparer<Double>.Default;
+
+        /// <summary>
+        ///
+        /// </summary>
+        public DoubleGenerator() : this(Double.MinValue, Double.MaxValue) {
+        }
+
+        /// <summary>
+        ///
+        /// </summary>
+        /// <param name="minimum"></param>
+        /// <param name="maximum"></param>
+        public DoubleGenerator(Double minimum, Double maximum) {
+            this.Minimum = minimum;
+            this.Maximum = maximum;
+
+            (this.defaultMinimumExponent, this.defaultMaximumExponent, this.defaultScales) =
+                CalculateScales(minimum, maximum);
+        }
+
+        /// <inheritdoc />
+        public Double Next() {
+            return Next(
+                Minimum,
+                Maximum,
+                this.defaultScales,
+                this.defaultMinimumExponent,
+                this.defaultMaximumExponent
+            );
+        }
+
+        /// <inheritdoc />
+        public Double NextDistinct(Double other) {
+            Double result;
+            do {
+                result = this.Next();
+            } while (EqualityComparer.Equals(other, result));
+
+            return result;
+        }
+
+        /// <inheritdoc />
+        public Double NextGreaterThan(Double other) {
+            // The minimum interval between floating point numbers gets larger as the values get
+            // larger.
+            var minInterval =
+                (Double)Math.Pow(
+                    10,
+                    Math.Max(EpsilonExponent, Math.Log10(Math.Abs(other)) - MinimumIntervalExponentDelta)
```

Comment:
I would include the how and the why of this opinionated implementation over a more naive `range * random.NextDouble()` implementation in a `<remarks></remarks>` section either at the top of the class.